### PR TITLE
[exporter/tanzuobservability] Don't roll resource tags into metrics

### PR DESCRIPTION
**Pull request by keep94**
[exporter/tanzuobservability] Don't roll resource tags into metrics
    
    Use the resourcetotelemetry helper instead.
    
    This involves several steps:
    1. attributesToTags() works like it did before. It simply concatenates multiple
    pdata.AttributeMap values into a single map[string]string
    2. Add newAttributeMap() to convert a map[string]string into a
    pdata.AttributeMap. This way you can still concatenate a map[string]string
    and a pdata.AttributeMap together into a single map[string]string because
    the traces exporter still needs to do this.
    3. Add getSource() function to get just the source from resource attributes.
    4. Add replaceSourceTag() function which replaces "source" with "_source"
    in the tags. This way the traces exporter can still work like it did before.
    5. Remove the Tags field from the metricInfo type so that it just has the
    metric and its source. We don't need to store the resource tags with metrics
    anymore.
    6. Update all attributesToTags() calls in the metrics exporter so that they
    get just the metric tags.
    7. Fix all metric exporter tests so that they no longer expect resource
    attributes in the metrics.
    8. Remove the tests that check that tags names "source" get renamed
    to "_source"
    9. Use the resourcetotelemetry helper.

**Description:** <Describe what has changed.>
Undo the rolling of resource tags into metrics. In the future we will rely on the resourcetotelemetry helper to do this for us.

**Testing:** <Describe what testing was performed and which tests were added.>
Unit tests.  Run OTEL collector with resourcetotelemetry helper turned off (default). Run OTEL collector with resourcetotelemetry helper turned on.

### CODE REVIEW

**Comment by keep94** [comment]
Hello, build-and-test unit test is failing, but I don't think the failure is related to the changes I made in this PR.  See the errors below. I don't recognize any of these tests:  

```=== PAUSE Test_PushMetrics/WAL/staleNaNIntGauge_case
=== RUN   Test_PushMetrics/WAL/staleNaNDoubleGauge_case
=== PAUSE Test_PushMetrics/WAL/staleNaNDoubleGauge_case
=== RUN   Test_PushMetrics/WAL/staleNaNIntSum_case
=== PAUSE Test_PushMetrics/WAL/staleNaNIntSum_case
=== RUN   Test_PushMetrics/WAL/staleNaNSum_case
=== PAUSE Test_PushMetrics/WAL/staleNaNSum_case
=== RUN   Test_PushMetrics/WAL/staleNaNHistogram_case
=== PAUSE Test_PushMetrics/WAL/staleNaNHistogram_case
=== RUN   Test_PushMetrics/WAL/staleNaNEmptyHistogram_case
=== PAUSE Test_PushMetrics/WAL/staleNaNEmptyHistogram_case
=== RUN   Test_PushMetrics/WAL/staleNaNSummary_case
=== PAUSE Test_PushMetrics/WAL/staleNaNSummary_case
=== CONT  Test_PushMetrics/WAL/doubleSum_case
=== CONT  Test_PushMetrics/WAL/emptyCumulativeSum_case
=== CONT  Test_PushMetrics/WAL/staleNaNSummary_case
=== CONT  Test_PushMetrics/WAL/staleNaNHistogram_case
=== CONT  Test_PushMetrics/WAL/staleNaNEmptyHistogram_case
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd1fd6c]
```

**Review by deosu — COMMENTED** [review]
Travis, the code changes look good to me. The only comment I can add here is that we are missing unit testing. Are there any in-direct tests we have, which are covering the code changes?

**Review by thepeterstone — COMMENTED** [review]
Looks good - a few minor comments. Thank you!

File: CHANGELOG.md
Comment (keep94) [reply CHANGELOG.md:20]: Nope. User can turn resourcetotelemetry on and off in config file.

File: exporter/tanzuobservabilityexporter/transformer.go
Comment (keep94) [reply exporter/tanzuobservabilityexporter/transformer.go:72]: orig.Attributes() could include a source tag, so we still need to replace source with _source.

**Review by thepeterstone — APPROVED** [review]

**Comment by keep94** [comment]
CHANGELOG.md changes very frequently, so I am constantly rebasing with main to resolve conflicts.  Is it possible to review this this PR even though this copy of CHANGELOG.md may have conflicts?

**Comment by keep94** [comment]
My build was all green before, but with my latest rebase to main, a load test is failing.  The only files with conflicts during my rebase were CHANGELOG.md and go.sum.  Could this test failure be unrelated to the changes made in this PR?

File: exporter/tanzuobservabilityexporter/metrics_test.go
Comment (oppegard) [inline exporter/tanzuobservabilityexporter/metrics_test.go:273]: I think we should inline this helper so that it's easy to see the hard-coded string values that are later used as assertions in the test.

**Review by oppegard — CHANGES_REQUESTED** [review]
Left one comment, but we'll pair-program on the full review and submit the changes.

File: exporter/tanzuobservabilityexporter/metrics_test.go
Comment (oppegard) [inline exporter/tanzuobservabilityexporter/metrics_test.go:41]: What's being expressed by having `source=my_source` on the metric and the resource? I thought that if a metric has an attribute named `source`, it would be renamed as `_source` in the transformed tobsMetric?

Comment (keep94) [reply exporter/tanzuobservabilityexporter/metrics_test.go]: The code under test doesn't know whether or not it is being wrapped by the resourcetotelemetry helper. By having "source" as both a resource attribute and a tag I am simulating the helper copying the resource attributes.  I reworked this test so that it tests both "source" -> "_source" and the deleting of the source resource attribute.

**Review by oppegard — APPROVED** [review]
Looks good – thanks Travis!

**Comment by keep94** [comment]
Hello.  My change has been approved internally, but I see that the load tests are failing.  These changes to our tanzuobservability exporter involve changing our unit tests. Can unit tests such as our unit tests for the tanzuobservability exporter affect the load tests?  It seems that the load tests are failing because too much RAM is being used.

**Comment by keep94** [comment]
Also, it seems that go.sum go.mod and changelog.md files change often.  Is it ok if I rebase once and for all after I have your approval?

**Comment by jpkrohling** [comment]
Can you rebase and fix the conflicts? I think the deb issue might also go away.

**Comment by keep94** [comment]
@jpkrohling I rebased my PR to the latest version of main to resolve the conflicts. The debian file built, but now the load tests are failing. Could these failures be flaky?

**Comment by jpkrohling** [comment]
Yes, those are intermittent failures. I rebased this to account for v0.48.0 changes.

**Comment by keep94** [comment]
Thank you @jpkrohling

**Comment by jpkrohling** [comment]
Looks like it has been affected by a flaky test. I just restarted the failed jobs, will merge on green.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
